pkg/ingester/ingester.go

```diff
@@ -719,40 +722,63 @@ type pushStats struct {
 	perMetricSeriesLimitCount int
 }
 
-// PushWithCleanup is the Push() implementation for blocks storage and takes a WriteRequest and adds it to the TSDB head.
-func (i *Ingester) PushWithCleanup(ctx context.Context, pushReq *push.Request) (*mimirpb.WriteResponse, error) {
-	// NOTE: because we use `unsafe` in deserialisation, we must not
-	// retain anything from `req` past the exit from this function.
-	defer pushReq.CleanUp()
-
+// StartPushRequest checks if ingester can start push request, and increments relevant counters.
+// If new push request cannot be started, errors converible to gRPC status code are returned, and metrics are updated.
+func (i *Ingester) StartPushRequest() error {
 	if err := i.checkRunning(); err != nil {
-		return nil, util_log.DoNotLogError{Err: err}
```

Comment:
Why was `DoNotLogError{}` removed?

Reply:
It was moved from `StartPushRequest` to `PushWithCleanup`.

When `StartPushRequest` is called from gRPC tap handle, no log messages will be logged anyway, because we do logging from interceptor, but interceptor will not run at all.

Reply:
> When StartPushRequest is called from gRPC tap handle, no log messages will be logged anyway, because we do logging from interceptor, but interceptor will not run at all.

I see. I think it's a bit obscure. Maybe we can improve the code comments to explain it?

Reply:
Will be fixed in https://github.com/grafana/mimir/pull/6073 (draft for now)